navigation.py

```diff
@@ -42,12 +42,40 @@ def navigate_to_link(link, driver):
     # wait for page to load, up to ten seconds
     WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, "//html")))
 
-def assert_current_page_is(page_name, driver):
-    if page_name in pages:
-        assert driver.current_url == pages[page_name]
-    else:
+def assert_current_page_is(page_name, driver, retry=0):
+    """ Assert that current url is as given.
+
+    Option arguments:
+        retry -- specifies how often to retry, with one second of delay between retries
+    """
+    if page_name not in pages:
         raise PageNotImplementedError(page_name, pages)
 
+    target_url = pages[page_name]
+    assert 0 <= retry
+    for i in range(retry):
+        if driver.current_url == target_url:
+            return
+        time.sleep(1)
+
+    if driver.current_url != target_url:
+        raise EndedUpOnWrongPageError(f"The current url {driver.current_url} does not match the expected url of page {page_name}")
+
+def assert_text_on_page(text, driver, retry=0):
+    """ Assert that given text shows up on page.
+
+    Option arguments:
+        retry -- specifies how often to retry, with one second of delay between retries
+    """
+    assert 0 <= retry
+    for i in range(retry):
+        if text in driver.page_source:
+            return
+        time.sleep(1)
+
+    if text not in driver.page_source:
+        raise TextNotFoundOnPageError(f"Text {text} was not found on current page {driver.current_url}")
```

Comment:
It seems you could extract common functionality from these two functions: e.g. a `waitUntil` method that takes a condition (as a closure), a number of retries, and an error to raise if the condition isn't eventually fulfilled. I don't know if this would end up being clearer, though.

Reply:
Sounds nice enough.